core/executor/src/wasm_runtimes_cache.rs

```diff
@@ -99,7 +99,12 @@ impl StateSnapshot {
 				// anyway.
 				let contents = mem::replace(segment.value_mut(), vec![]);
 
-				let init_expr = segment.offset().code();
+				let init_expr = match segment.offset() {
+					Some(offset) => offset.code(),
+					// Return if the segment is passive
+					None => return None
```

Comment:
I'm not sure what the correct behaviour here is.

Reply:
Yeah, I think this should do. Passive segments are only relevant to either bulk-memory ops or threading and they are both related to non-MVP features which we don't support yet.